packages/shell-api/src/modify-bson.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  ALL_PLATFORMS,
+  ALL_SERVER_VERSIONS,
+  ALL_TOPOLOGIES,
+  ServerVersions,
+  toStringMethods
+} from './enums';
+import Help from './help';
+/**
+ * This method modifies the BSON class passed in as argument. This is required so that
+ * we can have the driver return our BSON classes without having to write our own serializer.
+ * @param {Object} bson
+ */
+function constructHelp(className): Help {
+  const classHelpKeyPrefix = `shell-api.classes.${className}.help`;
+  const classHelp = {
+    help: `${classHelpKeyPrefix}.description`,
+    docs: `${classHelpKeyPrefix}.link`,
+    attr: []
+  };
+  return new Help(classHelp);
+}
+
+export default function(bson, platform): void {
+  [
+    'Binary', 'Code', 'DBRef', 'Decimal128', 'Int32', 'Long', /* 'Double',*/
+    'MaxKey', 'MinKey', 'ObjectId', 'Timestamp', 'Symbol'
+  ].forEach((className) => {
+    bson[className].prototype.serverVersions = ALL_SERVER_VERSIONS;
+    bson[className].prototype.platforms = ALL_PLATFORMS;
+    bson[className].prototype.topologies = ALL_TOPOLOGIES;
+
+    const help = constructHelp(className);
+    bson[className].prototype.help = (): Help => (help);
+    Object.setPrototypeOf(bson[className].prototype.help, help);
+  });
+
+  const toString = toStringMethods[platform];
```

Comment:
> Also with toStringMethods we make the platforms (CLI, browser) know something about themselves (how to inspect a BSON type) by the mean of the ShellApi, probably something that the platform should rather know directly.

Could you clarify this last point?

Reply:
we can also do something else in `modifyBson` to check for platform which is what is usually done in other tools that provide support for browser/node.js. It will not require us to pass in a platform at all:

```js
  if (typeof window !== 'undefined') {} // load browser specific requirement 
 
  try {
    // this will fail if we are not in node.js so we can put node.js specific toString method here
    require('util')
  } catch (e) {
    // browser + compass environment toString env can go here.
  }
```
([example Prior Art](https://github.com/choojs/on-performance/blob/master/index.js#L14-L25))

Reply:
> > Also with toStringMethods we make the platforms (CLI, browser) know something about themselves (how to inspect a BSON type) by the mean of the ShellApi, probably something that the platform should rather know directly.

> Could you clarify this last point?

What I meant is that the `toString` function is inherently a property of a platform (which `toString` to use depends on which platform). In the code that information however is in ShellApi. Since `modifyBson` is already invoked by Platform specific code then that information (`toString`) could be passed by the platform itself.

Reply:
I kinda like @lrlna's idea, it could be simple and effective. `require('util')` however won't throw cause of browserify, but maybe we could use that in our favor and just do directly here:

```
const toString = require('util').custom.inspect;
``` 

No need to deal with platforms as this should work everywhere we'd have a node.js `inspect` and do nothing if there is none. Probably we can't do better. Thoughts on Java? or they have a solution on their own?

Reply:
@mcasimir The downside of adding a second enum is that we already have the `platform` enum that we use for all other platform-specific decisions. Having two enums that are passed down, one for `platform` and one for the specific `toString` name, which are always going to be consistent, seems strange to me. Most of the enums are stored in shell-api, primarily because everything requires shell-api so it's the cleanest place to access them.

For @lrlna's idea, I think it's a bit odd to have a platform enum that we then ignore in order to do an ad-hoc check for if we're using Node. That may work for right now, where we are only dealing with node and not-node, but what about if the BSON package starts being used by Java? Or whatever other future platform?

Reply:
Hmmmm ok I feel ok to ignore the future platforms and then revisit it if we need. For now I'm happy to just have modify-bson do `require('util').inspect.custom || 'inspect'`

Reply:
Java I don't think uses these types, since they require the node bson types to be returned, but I will double check

Reply:
I think a good middle ground would be to add a asPrintable() method that can be used by other platforms if they need something other than util to print. This also brings the BSON classes in line with the rest of the shell API